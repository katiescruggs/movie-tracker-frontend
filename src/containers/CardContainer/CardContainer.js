import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Card from '../../components/Card/Card';
import './CardContainer.css';

export const CardContainer = ({movies, favorites, location}) => {
  const path = location.pathname;
  const displayArray = path === '/favorites' ? favorites : movies;
  
  const cards = displayArray.map((movie) => {
    return <Card
      key={movie.id}
      movie={movie} 
    />;
  });

  return (
    <div className='card-container'>
      {cards}
    </div>
  );
};

export const mapStateToProps = state => {
  return {
    movies: state.movies,
    favorites: state.favorites
  };
};

CardContainer.propTypes = {
  movies: PropTypes.array
};

export default connect(mapStateToProps, null)(CardContainer);