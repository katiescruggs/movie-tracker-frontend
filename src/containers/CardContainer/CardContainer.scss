.card-container {
  background-image: url('../../Assets/movie-background.jpg');
  background-position: center;
  background-size: cover;
  display: flex;
  overflow-x: scroll;
  height: calc(100vh - 100px);
}

.card-container::-webkit-scrollbar-track
{
  -webkit-box-shadow: inset 0 0 6px rgba(0,0,0,0.3);
  // background-color: #43465c;
}

.card-container::-webkit-scrollbar
{
  width: 6px;
  // background-color: #43465c;
}

.card-container::-webkit-scrollbar-thumb
{
  background-color: #B22628;
  border-radius: 20px;
}