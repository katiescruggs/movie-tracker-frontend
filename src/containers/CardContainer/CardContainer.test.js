import React from 'react'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import CardContainer from './CardContainer'

describe('CardContainer tests', () => {
  let CardContainer;
  beforeEach( () => {
    CardContainer = shallow(<Provider><CardContainer /></Provider>)
  })

  it('should be defined', () => {
    expect(CardContainer).toBeDefined()
  })

})