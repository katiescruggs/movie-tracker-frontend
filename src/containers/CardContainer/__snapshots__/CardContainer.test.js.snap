// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    favorites={
        Array [
            "array",
            "of",
            "favorites",
          ]
    }
    location={
        Object {
            "pathname": "current-path",
          }
    }
    movies={
        Array [
            "array",
            "of",
            "movies",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Component)
          movie="array"
          path="current-path"
/>,
        <Connect(Component)
          movie="of"
          path="current-path"
/>,
        <Connect(Component)
          movie="movies"
          path="current-path"
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Card: undefined",
        "nodeType": "class",
        "props": Object {
          "movie": "array",
          "path": "current-path",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Card: undefined",
        "nodeType": "class",
        "props": Object {
          "movie": "of",
          "path": "current-path",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Card: undefined",
        "nodeType": "class",
        "props": Object {
          "movie": "movies",
          "path": "current-path",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Component)
            movie="array"
            path="current-path"
/>,
          <Connect(Component)
            movie="of"
            path="current-path"
/>,
          <Connect(Component)
            movie="movies"
            path="current-path"
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Card: undefined",
          "nodeType": "class",
          "props": Object {
            "movie": "array",
            "path": "current-path",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Card: undefined",
          "nodeType": "class",
          "props": Object {
            "movie": "of",
            "path": "current-path",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Card: undefined",
          "nodeType": "class",
          "props": Object {
            "movie": "movies",
            "path": "current-path",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
